### PR TITLE
adjust some limitation about sharding ddl

### PR DESCRIPTION
**Pull request by lance6716**
### What is changed, added, or deleted? (Required)

In fact DM treats all sources in a task as a sharding group.

### Which DM version(s) do your changes apply to? (Required)

- [x] master (the latest development version, including v3.0 changes)
- [x] v2.0 (TiDB DM 2.0 versions)
- [ ] v1.0 (TiDB DM 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): close https://github.com/pingcap/dm/issues/1517

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by lance6716** [comment]
/cc @lichunzhu @GMHDBJD 

https://github.com/pingcap/dm/blob/a88019c97769c4bd7189cebcb73b812e4225ea26/dm/master/server.go#L133

we're using `server.getTaskResources`, which gets all sources of a task to form a sharding group

**Review by lichunzhu — APPROVED** [review]
/lgtm

File: zh/feature-shard-merge-pessimistic.md [hunk 1]
```diff
@@ -12,14 +12,12 @@ aliases: ['/docs-cn/tidb-data-migration/dev/feature-shard-merge-pessimistic/']
 
 DM 在悲观模式下进行分表 DDL 的迁移有以下几点使用限制：
 
+- 对于一个逻辑 sharding group，只能使用一个仅包含这些数据源的任务进行迁移。
+
 - 在一个逻辑 sharding group（需要合并迁移到下游同一个表的所有分表组成的 group）内，所有上游分表必须以相同的顺序执行相同的 DDL 语句（库名和表名可以不同），并且只有在所有分表执行完当前一条 DDL 语句后，下一条 DDL 语句才能执行。
```
Comment (CharLotteiu) [inline zh/feature-shard-merge-pessimistic.md:17]:
```suggestion
- 对于一个逻辑 sharding group（需要合并迁移到下游同一个表的所有分表组成的 group），只能使用一个仅包含这些数据源的任务进行迁移。

- 在一个逻辑 sharding group 内，所有上游分表必须以相同的顺序执行相同的 DDL 语句（库名和表名可以不同），并且只有在所有分表执行完当前一条 DDL 语句后，下一条 DDL 语句才能执行。
```

Does `这些数据源` refer to the data sources of all sharded tables in the sharding group?

File: en/feature-shard-merge-pessimistic.md [hunk 2]
```diff
@@ -11,10 +11,9 @@ This document introduces the sharding support feature provided by Data Migration
 
 DM has the following sharding DDL usage restrictions in the pessimistic mode:
 
+- For a sharding group, it is limited to use one task containing exactly the sources to perform the migration.
 - In a logical **sharding group** (composed of all sharded tables that need to be merged and migrated into one same downstream table), the same DDL statements must be executed in the same order in all upstream sharded tables (the schema name and the table name can be different), and the next DDL statement cannot be executed unless the current DDL operation is completely finished.
```
Comment (CharLotteiu) [inline en/feature-shard-merge-pessimistic.md:15]:
```suggestion
- For a logical **sharding group** (composed of all sharded tables that need to be merged and migrated into one same downstream table), it is limited to use one task containing exactly the sources to perform the migration.
- In a logical **sharding group** , the same DDL statements must be executed in the same order in all upstream sharded tables (the schema name and the table name can be different), and the next DDL statement cannot be executed unless the current DDL operation is completely finished.
```

File: zh/feature-shard-merge-pessimistic.md (hunk 1 above)
Comment (lance6716) [reply zh/feature-shard-merge-pessimistic.md:17]: “需要合并迁移到下游同一个表的分表所在数据源组成的 group” 
will change soon

Comment (lance6716) [reply zh/feature-shard-merge-pessimistic.md:17]: sharding group 还是指“分表”比较好。指数据源的时候表述为“sharding group 所在的数据源”。这样可以吗 @CharLotteiu @GMHDBJD

Comment (GMHDBJD) [reply zh/feature-shard-merge-pessimistic.md:17]: LGTM. sharding group 就是分表， sharding source/sharding group source 指数据源

Comment (CharLotteiu) [reply zh/feature-shard-merge-pessimistic.md:17]: LGTM

File: en/feature-shard-merge-pessimistic.md [hunk 3]
```diff
@@ -11,10 +11,9 @@ This document introduces the sharding support feature provided by Data Migration
 
 DM has the following sharding DDL usage restrictions in the pessimistic mode:
 
-- In a logical **sharding group** (composed of all sharded tables that need to be merged and migrated into one same downstream table), the same DDL statements must be executed in the same order in all upstream sharded tables (the schema name and the table name can be different), and the next DDL statement cannot be executed unless the current DDL operation is completely finished.
+- For a logical **sharding group** (composed of all sharded tables that need to be merged and migrated into one same downstream table), it is limited to use one task containing exactly the sources of shard tables to perform the migration.
```
Comment (CharLotteiu) [inline en/feature-shard-merge-pessimistic.md:14]:
```suggestion
- For a logical **sharding group** (composed of all sharded tables that need to be merged and migrated into one same downstream table), it is limited to use one task containing exactly the sources of sharded tables to perform the migration.
```

**Review by CharLotteiu — APPROVED** [review]
Rest LGTM

**Pull request merged**